### DISCUSSION TITLE
Render partial with block

### DISCUSSION THREAD

**Discussion by jbakerdev** [mojolicious/mojo#1726] [Q&A] [answered]
Within a template, how would one go about rendering a partial with a block?

Is there a better/easier way than this?

```
@@ index.html.ep
% layout 'application';
% title 'index';

<%= include 'partials/_card', title => 'Card #3', yield => begin %>
  <h1>Hello from Card #3.</h1>
<% end %>

@@ partials/_card.html.ep
<div class="card">
  <div class="card-body">
    <h5 class="card-title"><%= $title %></h5>
    <p class="card-text">
      <%= $yield->&* %>
    </p>
  </div>
</div>
```

**Comment by kraih** [comment, marked as answer]
Very creative, i never thought about using partial templates and blocks. But since there's nothing dynamic about the block, you can probably just use one of the content helpers to pass along the rendered result.

**Reply by jbakerdev** [reply]
Haha, well I can't take credit for this, it's a rails pattern that I use for sharing "wrapper" type partials across multiple views/templates (somewhat of a poor man's view component)

Perhaps I'm making this hard on myself, but after reading the docs I wasn't clear if the content/content_for/content_with helpers were the right choice for this. But since you mention them, would this be the "mojo way" of doing things?

```html
@@ index.html.ep

% content_with card_text => begin
  <p>text #1</p>
% end
%= include 'components/_card', card_title => 'title #1'

@@ partials/_card.html.ep
<div class="card">
  <div class="card-body">
    <h5 class="card-title"><%= $card_title %></h5>
    <p class="card-text"><%= content_with 'card_text' %></p>
  </div>
</div>
```

Or... do scrap the _card.html.ep partial, and embed that html in a custom helper that uses a block?

FWIW, this is what I was trying to emulate (abbreviated for brevity):

```html+erb
# index.html.erb
<%= render layout: ‘partials/card’, locals: { title => 'Card #3' } do %>
  <p>text #1</p> # Block
<% end %>

# partials/_card.html.erb
<div class="card">
  <div class="card-body">
    <h5 class="card-title"><%= title %></h5>
    <p class="card-text"><%= yield %></p>
  </div>
</div>
```

**Reply by kraih** [reply]
Yea, i'd say that's more the Mojo way to do it. Using `<p class="card-text"><%= content 'card_text' %></p>` in the partial template though.
